assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,145 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import { useSelect } from '@wordpress/data';
+import { ruleCard } from './styles';
+
+
+const Rule = ({
+  field,
+  loading,
+  newRule,
+  onAdd,
+  onDelete,
+  onUpdate,
+  reorderRule,
+  ruleIndex,
+  saving,
+  setOriginIndex,
+  setTargetIndex,
+  taxonomy,
+  term_id,
+  value,
+}) => {
+  const {
+    fields,
+    taxonomies,
+  } = wpLocalizedData;
+
+  const [rule, setRule] = useState({
+    field: Object.keys(fields)[0],
+    taxonomy: Object.keys(taxonomies)[0],
+    term_id: 0,
+    value: false,
+  });
+
+  // If existing rule, sync local state with incoming settings.
```

Comment:
Handle this by hoisting the add functionality to the main form. This component should get its values via props and not need to set defaults.